File: scripts/scripts/scripts/build-sitemap.mjs
// scripts/build-sitemap.mjs
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

// ★ あなたのPagesのベースURL（必要なら変更）
const baseUrl = 'https://s06659333-tech.github.io/niche';

const contentDir = path.resolve(__dirname, '..', 'content');
const outFile = path.resolve(__dirname, '..', 'sitemap.xml');

const entries = [];
entries.push(`${baseUrl}/`);
entries.push(`${baseUrl}/about.html`);

if (fs.existsSync(contentDir)) {
  const files = fs.readdirSync(contentDir)
    .filter(f => f.endsWith('.html'))
    .sort();

  for (const f of files) {
    // ファイル名に日本語が含まれるためURLエンコードを推奨
    entries.push(`${baseUrl}/content/${encodeURIComponent(f)}`);
  }
}

let xml = `<?xml version="1.0" encoding="UTF-8"?>\n`;
xml += `<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">\n`;
for (const loc of entries) {
  xml += `  <url><loc>${loc}</loc></url>\n`;
}
xml += `</urlset>\n`;

fs.writeFileSync(outFile, xml, 'utf-8');
console.log('sitemap.xml written with', entries.length, 'entries');

File: scripts/build_sitemap.py
# scripts/build_sitemap.py
import os, urllib.parse

BASE_URL = "https://s06659333-tech.github.io/niche"  # 必要なら変更
entries = [f"{BASE_URL}/", f"{BASE_URL}/about.html"]

if os.path.isdir("content"):
    files = sorted([f for f in os.listdir("content") if f.endswith(".html")])
    for f in files:
        entries.append(f"{BASE_URL}/content/{urllib.parse.quote(f)}")

xml = ['<?xml version="1.0" encoding="UTF-8"?>',
       '<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">']
xml += [f"  <url><loc>{loc}</loc></url>" for loc in entries]
xml.append('</urlset>')

with open("sitemap.xml","w",encoding="utf-8") as fp:
    fp.write("\n".join(xml) + "\n")

print("sitemap.xml written with", len(entries), "entries")
